packages/next/client/image.tsx

```diff
@@ -400,7 +400,7 @@ function getInt(x: unknown): number | undefined {
 
 function defaultImageLoader(loaderProps: ImageLoaderPropsWithConfig) {
   const loaderKey = loaderProps.config?.loader || 'default'
-  const load = loaders.get(loaderKey)
+  const load = loaders.get(loaderKey as LoaderValue)
```

Comment:
I think we need to solve this a different way. We need to pick one of the two solutions:

1. This new context feature should work for both `next/image` and `next/future/image`
2. OR this new context feature should only work for `next/future/image` in which case, we should not change the types of `next/image`